File: app/src/main/java/com/mxt/anitrend/data/converter/ImageBaseConverter.java
package com.mxt.anitrend.data.converter;

import com.mxt.anitrend.model.api.retro.WebFactory;
import com.mxt.anitrend.model.entity.anilist.meta.ImageBase;

import io.objectbox.converter.PropertyConverter;

public class ImageBaseConverter implements PropertyConverter<ImageBase, String> {

    @Override
    public ImageBase convertToEntityProperty(String databaseValue) {
        if(databaseValue == null)
            return null;
        return WebFactory.gson.fromJson(databaseValue, ImageBase.class);
    }

    @Override
    public String convertToDatabaseValue(ImageBase entityProperty) {
        if(entityProperty == null)
            return null;
        return WebFactory.gson.toJson(entityProperty);
    }
}

File: app/src/main/java/com/mxt/anitrend/data/converter/ImageBaseConverter.kt
package com.mxt.anitrend.data.converter

import com.mxt.anitrend.model.api.retro.ServiceFactory
import com.mxt.anitrend.model.entity.anilist.meta.ImageBase
import io.objectbox.converter.PropertyConverter

class ImageBaseConverter : PropertyConverter<ImageBase, String> {
    override fun convertToEntityProperty(databaseValue: String?): ImageBase? = databaseValue?.let { ServiceFactory.gson.fromJson(it, ImageBase::class.java) }

    override fun convertToDatabaseValue(entityProperty: ImageBase?): String? = entityProperty?.let { ServiceFactory.gson.toJson(it) }
}
